src/jimgw/single_event/transforms.py

```diff
@@ -0,0 +1,231 @@
+import jax.numpy as jnp
+from beartype import beartype as typechecker
+from jaxtyping import Float, Array, jaxtyped
+from astropy.time import Time
+
+from jimgw.single_event.detector import GroundBased2G
+from jimgw.transforms import BijectiveTransform, NtoNTransform
+from jimgw.single_event.utils import (
+    m1_m2_to_Mc_q,
+    Mc_q_to_m1_m2,
+    m1_m2_to_Mc_eta,
+    Mc_eta_to_m1_m2,
+    q_to_eta,
+    eta_to_q,
+    ra_dec_to_zenith_azimuth,
+    zenith_azimuth_to_ra_dec,
+    euler_rotation,
+    spin_to_cartesian_spin,
+)
+
+
+@jaxtyped(typechecker=typechecker)
+class ComponentMassesToChirpMassMassRatioTransform(BijectiveTransform):
+    """
+    Transform chirp mass and mass ratio to component masses
+
+    Parameters
+    ----------
+    name_mapping : tuple[list[str], list[str]]
+            The name mapping between the input and output dictionary.
+    """
+
+    def __init__(
+        self,
+        name_mapping: tuple[list[str], list[str]],
```

Comment:
For the, let's call them "pre-made" transformations such as the masses transforms for which the transforms are already in the source code, the `name_mapping` argument is not needed and could be removed from the `__init__` altogether. This would also avoid having to check the assertions, which could reduce the number of errors made in the code and make the code less confusing. Practically speaking, I think this would imply that `name_mapping` should become an optional argument of transform, but that might lead to unexpected behaviors for other types of transforms. I am not sure if there is an easy way to get the best of both worlds...

Reply:
@ThibeauWouters Sorry for the late reply. I also thought of the implementation you mentioned. Maybe we can discuss this in the Monday call and see which implementation people like more. I think the benefit of the current implementation is that user can keep track of the naming when they are setting the `sample_transforms`.
```
sample_transforms = [
    ComponentMassesToChirpMassMassRatioTransform(name_mapping=[["m_1", "m_2"], ["M_c", "q"]]),
    BoundToUnbound(name_mapping = [["M_c"], ["M_c_unbounded"]], original_lower_bound=M_c_min, original_upper_bound=M_c_max),
    BoundToUnbound(name_mapping = [["q"], ["q_unbounded"]], original_lower_bound=q_min, original_upper_bound=q_max),
    ...
```
Users can see that `m_1` and `m_2` are transformed to `M_c_unbounded` and `q_unbounded` in this case.

I also thought of another way that is to make `__init__` taking four arguments, each specifying a name. Users can keep the freedom to set any name in that case.
```
def __init__(m1_name, m2_name, Mc_name, q_name):
    ...
```